### PR TITLE
Add release notes for 2.10.0.0

### PR DESCRIPTION
**Pull request by peternied**
### Description
Add release notes for 2.10.0.0

### Check List
- [ ] New functionality includes testing
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peternied** [comment]
Preparing the list of release notes in advance of the 2.10 release [1], if any additional features make it before code freeze they will be included.

- [1] #3028

**Comment by cwperks** [comment]
@peternied Please mark this and the security-dashboards-plugin ready for review when this is ready to be reviewed.

**Review by cwperks — APPROVED** [review]
@peternied Looks good to me. Should https://github.com/opensearch-project/security/pull/3330 also be included here?

**Review by cwperks — DISMISSED** [review]
@peternied Looks good to me. Should https://github.com/opensearch-project/security/pull/3330 also be included here?

File: release-notes/opensearch-security.release-notes-2.10.0.0.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+* Switch CodeQL to assemble artifacts using the same build as the rest of CI ([#3132](https://github.com/opensearch-project/security/pull/3132))
+* Only run the backport job on merged pull requests ([#3134](https://github.com/opensearch-project/security/pull/3134))
+* Add code coverage exclusions on false positives ([#3196](https://github.com/opensearch-project/security/pull/3196))
+* Enable jarhell check ([#3227](https://github.com/opensearch-project/security/pull/3227))
+* Retry code coverage upload on failure ([#3242](https://github.com/opensearch-project/security/pull/3242))
+* [Refactor] Adopt request builder patterns for SecurityRestApiActions for consistency and clarity ([#3123](https://github.com/opensearch-project/security/pull/3123))
+* [Refactor] Remove json-path from deps and use JsonPointer instead ([#3262](https://github.com/opensearch-project/security/pull/3262))
+* Use version of org.apache.commons:commons-lang3 defined in core ([#3306](https://github.com/opensearch-project/security/pull/3306))
+* Fix checkstyle #3283
```
Comment (scrawfor99) [inline release-notes/opensearch-security.release-notes-2.10.0.0.md:75]:
```suggestion
* Fix checkstyle #3283
* Demo Configuration changes ([#3330](https://github.com/opensearch-project/security/pull/3330))
```

**Review by RyanL1997 — APPROVED** [review]

**Comment by cwperks** [comment]
I will update this PR so that it matches the one in the 2.10 branch: https://github.com/opensearch-project/security/pull/3337

**Pull request closed without merging**